service/application/src/main/java/com/solace/maas/ep/event/management/agent/cli/EmaCommandLine.java

```diff
@@ -0,0 +1,210 @@
+package com.solace.maas.ep.event.management.agent.cli;
+
+import com.solace.maas.ep.event.management.agent.repository.model.mesagingservice.MessagingServiceEntity;
+import com.solace.maas.ep.event.management.agent.repository.model.scan.ScanStatusEntity;
+import com.solace.maas.ep.event.management.agent.scanManager.ScanManager;
+import com.solace.maas.ep.event.management.agent.scanManager.model.ScanRequestBO;
+import com.solace.maas.ep.event.management.agent.scanManager.model.ZipRequestBO;
+import com.solace.maas.ep.event.management.agent.service.DataCollectionFileService;
+import com.solace.maas.ep.event.management.agent.service.ImportService;
+import com.solace.maas.ep.event.management.agent.service.MessagingServiceDelegateServiceImpl;
+import com.solace.maas.ep.event.management.agent.service.ScanStatusService;
+import com.solace.maas.ep.event.management.agent.util.IDGenerator;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Component
+@Order(1)
+@Slf4j
+public class EmaCommandLine implements CommandLineRunner {
+    private final ScanManager scanManager;
+    private final ImportService importService;
+    private final ScanStatusService scanStatusService;
+    private final IDGenerator idGenerator;
+    private final DataCollectionFileService dataCollectionFileService;
+    private final MessagingServiceDelegateServiceImpl messagingServiceDelegateService;
+
+    public EmaCommandLine(ScanManager scanManager,
+                          ImportService importService,
+                          ScanStatusService scanStatusService,
+                          IDGenerator idGenerator,
+                          DataCollectionFileService dataCollectionFileService,
+                          MessagingServiceDelegateServiceImpl messagingServiceDelegateService) {
+        this.scanManager = scanManager;
+        this.importService = importService;
+        this.scanStatusService = scanStatusService;
+        this.idGenerator = idGenerator;
+        this.dataCollectionFileService = dataCollectionFileService;
+        this.messagingServiceDelegateService = messagingServiceDelegateService;
+    }
+
+    @Override
+    public void run(String... args) throws Exception {
+        if (args.length > 2) {
+
+            String type = args[0];
+
```

Comment:
For now, only handle the scan command.

Reply:
comment for next ticket, let's change this var to `commandType` or `command` or something else.